Bottom: 72db6fd5f6da36c500e75eeab8e7d57b8bc60834
Top:    98d4ca9a26635ddb0c15dbb599a717a151308091
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-05-25 12:23:10 -0700

acm: Use -1 to indicate an invalid socket rather than 0

socket() can return 0 as a valid socket.  This can happen
when using a daemon that closes stdin/out/err.

Signed-off-by: Sean Hefty <sean.hefty@intel.com>


---

diff --git a/src/acm.c b/src/acm.c
index bcf11da..9c65919 100755
--- a/src/acm.c
+++ b/src/acm.c
@@ -61,7 +61,7 @@ typedef struct acm_msg cma_acm_msg_t;
 #endif
 
 static pthread_mutex_t acm_lock = PTHREAD_MUTEX_INITIALIZER;
-static int sock;
+static int sock = -1;
 static short server_port = 6125;
 
 struct ib_connect_hdr {
@@ -108,12 +108,12 @@ void ucma_ib_init(void)
 
 err:
 	close(sock);
-	sock = 0;
+	sock = -1;
 }
 
 void ucma_ib_cleanup(void)
 {
-	if (sock > 0) {
+	if (sock >= 0) {
 		shutdown(sock, SHUT_RDWR);
 		close(sock);
 	}
@@ -322,7 +322,7 @@ void ucma_ib_resolve(struct rdma_addrinfo **rai, struct rdma_addrinfo *hints)
 	struct acm_ep_addr_data *data;
 	int ret;
 
-	if (sock <= 0)
+	if (sock < 0)
 		return;
 
 	memset(&msg, 0, sizeof msg);
